File: SwiftUI/TheHotnessSwiftUI/BGGApi/Game.swift
//
//  Game.swift
//  TheHotnessSwiftUI
//
//  Created by Petteri Kamppuri on 28.9.2020.
//

import Foundation

struct Game: Decodable {
    let name: String
    let id: String
    let rank: Int
    let thumbnailUrl: URL
    let yearPublished: String
}

File: Flutter/lib/BGGApi/Game.dart
class Game {
  final String id;
  final String name;
  final String yearPublished;
  final Uri thumbnailUrl;

  Game({this.id, this.name, this.yearPublished, this.thumbnailUrl});

  factory Game.fromJson(Map<String, dynamic> json) {
    return Game(
      id: json['id'],
      name: json['name'],
      yearPublished: json['yearPublished'],
      thumbnailUrl: Uri.parse(json['thumbnailUrl']),
    );
  }
}
